Ical.Net/Collections/ExceptionDateCollection.cs

```diff
@@ -0,0 +1,66 @@
+﻿//
+// Copyright ical.net project maintainers and contributors.
+// Licensed under the MIT license.
+//
+
+#nullable enable
+using System.Collections.Generic;
+using Ical.Net.DataTypes;
+
+namespace Ical.Net.Collections;
+
+/// <summary>
+/// Represents a collection of exception dates for calendar events.
+/// <para>
+/// This class is used to manage dates that should be excluded from recurring events.
+/// </para>
+/// <para>
+/// The main feature of this class is the <see cref="ToExceptionDates"/> method, which aggregates
+/// and converts the exception <see cref="CalDateTime"/> objects into a list of <see cref="PeriodList"/> objects.
+/// This method ensures that the periods are grouped by their timezone IDs and period kinds, and that
+/// each <see cref="PeriodList"/> contains only distinct periods.
+/// </para>
+/// </summary>
+public class ExceptionDateCollection : PeriodCollectionBase
```

Comment:
Wouldn't it be possible (and cleaner) letting the `ExceptionDateCollection` be a collection of type `IDateTime` rather than `Period`? In case of `RDATE` we need to deal with the different type of entries but in case of EXDATEs this isn't necessary, so it would probably be cleaner to have it strongly typed. Also see my comment in `PeriodCollectionBase`, maybe replace the whole class with just `List<IDateTime>` and implement the specific functionality as extension methods?

Reply:
Oh, just realizing that the class is only used in test cases. Shouldn't we move this to the test projects then?